README.md
# The Sentinel's Lexicon
Hi there, and welcome!

This project is intended to be a megawad complation that will contain some of
the most pouplar and fun mapsets. This will also contain short mapsets also.

This is to be used for Zandronum.

It will also contain a hub map where players can vote on the mapset to play on.

## What mapsets are in there so far?
So far, the mapsets included are:

- Alien Vendetta
- Combat Shock 1 and 2
- Hell Revealed 1 and 2
- Kamasutra
- New Gothic Movement 1
- Shai'tans Luck
- Speed of Doom
- Dark Tartarus
- VanGaurd
- Scythe 2
- Whispers of Satan
- UAC Ultra
- Monuments of Mars
- Khorus's Speedy shit
- Forest Swords
- Circle of Caina
- Doom Core Trilogy
- Maps of Chaos
- Drown in Blood
- Swift Death
- Going Down
- Estranged
- Dark Encounters
- Hadephobia
- Deus Vult
- Mayhem 17
- Sunlust
- Pizza Steve
- Hellbound
- 1994 Tune up Community Project
- Unholy Realms
- Hellcore 2.0

We also started to add single map wads. So far, these are included:

- The Spire
- The Eye

You can also access doom 2 from the hub now!

## How to build the pk3?
Clone this repo to your machine and then run CompilePk3.bat

Automatic builds can be found at https://jenkins.csnxs.uk/job/Lexicon/

## Where can I play this?
This can be played and hosted on [The Sentinel's Playground](https://allfearthesentinel.net "The Sentinel's Playground") with builds that i release.

I would like to ask that you do not upload builds compiled by yourself to the service as I will be also uploading any builds that i deem to be released as alpha or beta etc there. I will actually tell sgt Nax on you if you do this.

You are however free to upload your own compiled build as a private wad.

## Want to suggest other mapsets or report bugs?
Feel free to use the tracker on here. You can also join our discord at https://discord.gg/qj9GASW

## Credits
- MiFU for doing most of the compile work, its a pain in the ass.
- K0RPSE for also assisting with the compile work, its still a pain in the ass though.
- Shaewn for assisting with various issues. He should compile more shit though but thats just me.
- Tribeam for scripting, effects, VR texturing.
- Fiendzy vr map design.
- Michaelis for adding more playerstarts to maps.
- Our testers (Kenopath, Blushberry)
- Various map authors for the fun maps that are included in this pack. I'll get a detailed list later. FYI, none of the maps have changed, just map names and texture renames to ensure nothing conflicts and more playerstarts to better support for multiplayer
- Sega for the Sonic hud font, HD version and extra chars made by Tribeam.
- Skulltag for the railgun sprites used for the InstaKiller
- Guilty Gear for the hub music
- Corgiatom for the hub music
- Willo for some of the artwork included in the HUB map
- and YOU, for playing the damn thing. Cant do it without YOU

Changelog.txt
Stuff:
2b09b9e Fixed texture bugs im D2TWID on map01 Also added a lill gift for Fookerton
4efc496 Fixed errors with credits script not running
5e859e9 Finished credits also made hub music random when entering hub
be8682b Revert "Added Mayhem17 textures"
604ae22 Revert "Added hudmessage at map start, showing the current mapset, level, and credits for said map. Please see the LANGUAGE lump for info on how to add credits."
92cb4fe Added Mayhem17 textures
210524e Added hudmessage at map start, showing the current mapset, level, and credits for said map. Please see the LANGUAGE lump for info on how to add credits.
a5447a6 Update README.md
a0d6241 Changed the music.
a2a087d Removed "if all players have voted, end countdown early" functionality. This allows users to change their minds, or allow more players to join and get a chance to vote.
fbbfaff Added endoom lump
6fe3d56 Added 2 new screens, highlighted the discord link, fixed missing char from font.
f966608 Added drown in blood and Swift Death to HUB. DID is next to Sunder. Switft death is facing combat shock 1
0fdecc6 fixed swift death interpic
c6f98a8 Added swift death Maps
002efdb Added swift death Music
92af5bd Added swift death GFX
4718e6a Added swift death mapinfo
635a23c Updated mapinfo
f037035 Fixed switch texture issues for Drown in blood
992c4c2 Fixed switch texture issues for Drown in blood
4af762d Added rest of the DIB files and added mapinfo
3bd6043 Added DIB flats
df494dd Added Drown in blood patches
de5f139 HUB: Added 2 new screens. changed the fast compiler to not do any compression at all, as it is now too slow because the mod is too damn big :D
1f0f22f Added Doom 2 the way iD did and Sunder. D2TWID is located next to maps of chaos while sunder is next to Dark tartarus
a2f7651 Ignore the memes
3c3e8ee Ignore these memes
b231794 Added maps dor d2twid
a3d2164 Added the rest of the shit
8111ec4 Added D2TWID flats
022c3c1 Added D2TWID Textures
0ae61d0 Added D2TWID sounds
895c8f5 Added D2TWID sprites and mapinfo/defs
b02f038 Missing music
8c6fae9 Added Sunder
a7647c7 Update README.md
77535f8 Merge branch 'master' of https://dev.allfearthesentinel.net/mifu/lexicon
cfa2c16 Fixed maps also on forest swords to use new names for the slime flats
f6b5b03 Had to rename slime textures for forest swords. fuck you complex doom
4c6b2a5 Edited file names to ensure its a .png
9553998 Added 2 more screens
dfdd1b1 Removed music replacing doom 2 music
451c87c fixed mapinfo and acs
e90fd32 Added and completed Doom Triology (now added to hub) and Maps of Chaos. Also renamed the WOS CWIL gfx
1d58a29 Added a rather "special sound"
4b700c6 Removed duplicate
7a3361f Fixed a texture up
feeea6e Merge branch 'master' of https://dev.allfearthesentinel.net/mifu/lexicon
98be738 Added Doom Core Trilogy maps and textures
ab02f33 Update README.md
2e1ddd9 Update README.md
f545f0f Hub: Added screen for Forest Swords.
313340b Added Forest Swords to the HUB. The switch is next to Circle of Caina
6e4fc50 Fixed some textures
38a8ea9 Moved Dark Tartarus textures to textures folder
9544f26 Fixed Grass texture for forrest swords
ab86035 Fixed Grass texture for forrest swords
ca2d646 Fixed Grass texture for forrest swords
8b87ae5 Moved Dark Tartarus textures to textures folder
0bbdcef Removed dupilcate and re-added problematic texture
9b4689f Renamed texture files
62a34eb Added Forest Swords resources, maps and music and edited decorate and mapinfo.
f9a948e Renamed some textures as lump names were too big. Added some Forest Swords sprites.
5ddc629 HUB: Added screen for Circle of Caina.
e76c972 Fixed broken textures in COC03 and 04
242667a Edited HUB to add Circle of Caina. It's next to Monuments of Mars
f7ecc42 Merge branch 'doomer' into 'master'
e8bd7a8 Whoops
6824fd5 Fixed mapinfo
84c59d9 Merge branch 'doomer' of https://dev.allfearthesentinel.net/mifu/lexicon into doomer
96e514a Added Circle of Caina maps and applied a fix for fog in uac ultra 1.
2379349 Merge branch 'master' into 'doomer'
102d19c Update MAPINFO.txt to fix HR07. Needed map07special
c02637b HUB: turned off the lights on the unused switches.
a1da906 Merge branch 'doomer' of https://dev.allfearthesentinel.net/mifu/lexicon into doomer
072f163 Added Circle of Caina assets and added Decorate actors and music
8985119 Merge branch 'master' into 'doomer'
c5bd57e Update README.md
d6d68a9 Update README.md
ac15af8 Fixed Alignment of particle sprites.
6344f92 Whoops, left frame duration as 0, causing the game to crash.
58ff6cb Tweaked 'debug mode' text position; InstaKiller priority is now 1, InstaKiller no longer has a spread.
3a558b9 Removed the kill monsters cheat, having both the InstaKiller and the Kill Monsters cheat is redundent.
e0be94a Moved the InstaKiller to slot 0.
c6f01f6 Added debug switches: God mode and Insta Killer; The Insta Killer is an insanely overpowered weapon, uses the ST railgun sprites, and a custom particle effect by me.
8d7c6ac Fixed debug cvar default value.
524659a Added debug mode message.
1a3f50e Added screen for KSS.
643a3e9 Merge branch 'doomer' into 'master'
84f4593 I pulled a tribeam and forgot to save the hub
5eed4d1 Merge branch 'doomer' into 'master'
27cb75b Renamed music for KSS and edited hub map. Switch for KSS is behind vanguard
a92a80a Added Khorus's Speedy Shit maps, mapinfo and music
f9ceafc Merge branch 'master' into 'doomer'
f5485d6 Merge branch 'master' of https://dev.allfearthesentinel.net/mifu/lexicon
2b2d67f HUD: Added debug mode cvar. Added switches for debug mode cheats(they dont function yet). Switches are only opened when 'lexicon_debug_mode' equals 1
54da404 Update README.md
1552d3d HUB: Made static animation faster. Changed off air screens to static. All switches with static now do nothing.
4ea5553 HUB: Fixed Shai'tan's luck screen; made Speed of Doom pic brighter.
a389def HUB: Added screen for Monuments of Mars
75e56e1 Merge branch 'doomer' into 'master'
eedc90a Edited ACS and added slot 14 to one of the switches (near the rest of the normal maps)

pk3/Acs/Lexicon.c
// written in BCS
#include "../../compiler/lib/zcommon.h"
#library "vote"

strict namespace
{
    #if 1
        #define STATE_INIT -1
        #define STATE_VOTEWAIT 0
        #define STATE_COUNTDOWN 1
        #define STATE_CHECKTIE 2
        #define STATE_RESULTS 3

        #define MAPSET_MAX 192
        #define MAPSET_SECTIONS 3
        #define MAPSET_SECTIONS_MAX 64

        #define PLAYER_MAX 64
    #endif

    /////////////////////
    // vote manager
    /////////////////////
    int votes[MAPSET_MAX];              // holds the votes
    int votessorted[MAPSET_MAX][2];     // all the votes, sorted
    int votecount = 0;                  // amount of votes made

    int time_ticks = 0;                 // the time left in ticks
    int time_seconds = 0;               // the time left in seconds

    int players[PLAYER_MAX];            // all the player info

    int state = STATE_INIT;             // state of the voting system
    int state_clock;                    // custom timer

    /////////////////////
    // hud
    /////////////////////
    fixed hud_width;                    // the position of the right of the screen
    fixed hud_height;                   // the position of the bottom of the screen
    fixed hud_width_half;               // the position of the center of the screen on the x axis
    fixed hud_height_half;              // the position of the center of the screen on the y axis

    struct obj_confetti
    {
        int x;
        int y;
        int velx;
        int vely;
        int animnum;
        int confnum;
        str image;
    };
    struct obj_confetti objs_confetti[128];

    // section names
    str sectionnames[MAPSET_SECTIONS];

    /////////////////////
    // level
    /////////////////////
    int levelstarted = 0;
    int clock = 0;
    int countstart = 0;
    str clockcolor = "\c[Green]";

    // this is where the names that show up in the votes are, and the map names to go to when that wad wins
    str votenames[MAPSET_MAX+1][2] =
    {
        // mapset name                       // the map this vote will take players too
																	// the number a linedef uses to cast the vote
        // Normal mapsets
        { "Nothing",                        "----"          }, 		// 0
        { "Hell on Earth",                  "MAP01"         }, 		// 1
        { "Hell Revealed",                  "HR01"          }, 		// 2
        { "Hell Revealed 2",                "HR2_01"        }, 		// 3
        { "Kamasutra",                      "KS01"          }, 		// 4
        { "Shai'tans Luck",                 "SL20"          }, 		// 5
        { "Speed Of Doom",                  "SOD01"         }, 		// 6
        { "Vanguard",                       "VAN01"         }, 		// 7
        { "Scythe 2",                       "SC201"         }, 		// 8
		{ "Whispers Of Satan",		        "WOS01"	        }, 		// 9
		{ "UAC Ultra",				        "UAC01"	        }, 		// 10
		{ "Monuments Of Mars",		        "MOM01"	        }, 		// 11
		{ "Khorus's Speedy Shit",	        "KSS01"	        }, 		// 12
		{ "Circle of Caina",		        "COC01"	        }, 		// 13
		{ "Forest Swords",			        "FSW01"	        }, 		// 14
		{ "Doom Core Trilogy",		        "DCOR01"	    }, 		// 15
		{ "Maps of Chaos",	                "MOC01"	        }, 		// 16
		{ "Doom2 The Way ID Did",	        "WID01"         }, 		// 17
		{ "Estranged",				        "EST01"         }, 		// 18
		{ "Going Down",				        "GD01"	        }, 		// 19
		{ "Dark Encounters",		        "DKE01"	        }, 		// 20
        { "Alien Vendetta",                 "AV__01"        }, 		// 21
		{ "Hadephobia",				        "HPH01"         }, 		// 22
		{ "Mayhem 17",				        "MAY01"	        }, 		// 23
		{ "Sunlust",				        "SLU01"	        }, 		// 24
		{ "Hellbound",				        "HLB01"         }, 		// 25
		{ "1994 Tune Up Community Project", "94TU01"        }, 		// 26
		{ "Unholy Realms",			        "UHR01"	        }, 		// 27
        { "Hell Core",			            "HC01"	        }, 		// 28
        { "Ancient Aliens",			        "AA101"	        }, 		// 29
        { "Valiant",			            "VAL01"	        }, 		// 30
        { "Epic",			                "EP101"	        }, 		// 31
        { "Epic 2",			                "EP201"	        }, 		// 32
        { "Nova",			                "NV101"	        }, 		// 33
        { "Community Chest 1",			    "CC101"	        }, 		// 34
        { "Community Chest 2",			    "CC201"	        }, 		// 35
        { "Community Chest 3",			    "CC301"	        }, 		// 36
        { "Community Chest 4",			    "CC401"	        }, 		// 37
        { "Stardate 20x6",			        "SD601"	        }, 		// 38
        { "Stardate 20x7",			        "SD701"	        }, 		// 39
        { "Swim With The Whales",			"SW101"	        }, 		// 40
        { "Congestion 1024",			    "TT101"	        }, 		// 41
        { "Clausterphobia 2",			    "TT201"	        }, 		// 42
        { "Clausterphobia",			        "TT301"	        }, 		// 43
        { "Chainworm",			            "CW101"	        }, 		// 44
        { "Back to Saturn X - EP1",			"BTSX01"	    }, 		// 45
        { "Zones Of Fear",			        "ZOF01"	        }, 		// 46
        { "Zone 300",			            "ZTH01"	        }, 		// 47
        { "Interception",			        "INT01"	        }, 		// 48
        { "Doom2 Redux",			        "REDX01"	    }, 		// 49
        { "*Knee-Deep in the Dead",		    "E1M1"	        }, 		// 50
        { "*The Shores of Hell",		    "E2M1"	        }, 		// 51
        { "*Inferno",			            "E3M1"	        }, 		// 52
        { "*Thy Flesh Consumed",			"E4M1"	        }, 		// 53
        { "*Sigil",			                "E5M1"	        }, 		// 54
        { "*No Rest For The Living",	    "NV_MAP01"	    }, 		// 55
        { "*The Master Levels",			    "ML_MAP01"	    }, 		// 56
        { "*TNT: Evilution",			    "TN_MAP01"	    }, 		// 57
        { "*The Plutonia Experiment",		"PL_MAP01"	    }, 		// 58
        { "Unused",			                "MAP01"	        }, 		// 59
        { "Unused",			                "MAP01"	        }, 		// 60
        { "Unused",			                "MAP01"	        }, 		// 61
        { "Unused",			                "MAP01"	        }, 		// 62
        { "Unused",			                "MAP01"	        }, 		// 63
        { "Unused",			                "MAP01"	        }, 		// 64

        // Slaughter Mapsets
        { "Combat Shock",                   "CS01"             }, 	// 65
        { "Combat Shock 2",                 "CS201"            }, 	// 66
        { "New Gothic Movement 1",          "NG101"            }, 	// 67
        { "Dark Tartarus",                  "TAT01"            }, 	// 68
		{ "Sunder",					        "SND01"            }, 	// 69
		{ "Drown in Blood",			        "DIB01"	           }, 	// 70
		{ "Swift Death",			        "SDE01"            }, 	// 71
        { "Deus Vult",				        "DV01"	           }, 	// 72
        { "Pizza Steve",			        "PIZ01"            }, 	// 73
        { "Deus Vult 2",			        "DV201"            }, 	// 74
        { "SlaughterFest 2012",			    "SF201"            }, 	// 75
        { "New Gothic Movement 2",			"NG201"            }, 	// 76
        { "SlaughterFest 2013",			    "SF301"            }, 	// 77
        { "Chillax",			            "CHX01"            }, 	// 78
        { "Destination Unknown",			"DU101"            }, 	// 79
        { "Unused",			                "MAP01"            }, 	// 80
        { "Unused",			                "MAP01"            }, 	// 81
        { "Unused",			                "MAP01"            }, 	// 82
        { "Unused",			                "MAP01"            }, 	// 83
        { "Unused",			                "MAP01"            }, 	// 84
        { "Unused",			                "MAP01"            }, 	// 85
        { "Unused",			                "MAP01"            }, 	// 86
        { "Unused",			                "MAP01"            }, 	// 87
        { "Unused",			                "MAP01"            }, 	// 88
        { "Unused",			                "MAP01"            }, 	// 89
        { "Unused",			                "MAP01"            }, 	// 90
        { "Unused",			                "MAP01"            }, 	// 91
        { "Unused",			                "MAP01"            }, 	// 92
        { "Unused",			                "MAP01"            }, 	// 92
        { "Unused",			                "MAP01"            }, 	// 94
        { "Unused",			                "MAP01"            }, 	// 95
        { "Unused",			                "MAP01"            }, 	// 96
        { "Unused",			                "MAP01"            }, 	// 97
        { "Unused",			                "MAP01"            }, 	// 98
        { "Unused",			                "MAP01"            }, 	// 99
        { "Unused",			                "MAP01"            }, 	// 100
        { "Unused",			                "MAP01"            }, 	// 101
        { "Unused",			                "MAP01"            }, 	// 102
        { "Unused",			                "MAP01"            }, 	// 103
        { "Unused",			                "MAP01"            }, 	// 104
        { "Unused",			                "MAP01"            }, 	// 105
        { "Unused",			                "MAP01"            }, 	// 106
        { "Unused",			                "MAP01"            }, 	// 107
        { "Unused",			                "MAP01"            }, 	// 108
        { "Unused",			                "MAP01"            }, 	// 109
        { "Unused",			                "MAP01"            }, 	// 110
        { "Unused",			                "MAP01"            }, 	// 111
        { "Unused",			                "MAP01"            }, 	// 112
        { "Unused",			                "MAP01"            }, 	// 113
        { "Unused",			                "MAP01"            }, 	// 114
        { "Unused",			                "MAP01"            }, 	// 115
        { "Unused",			                "MAP01"            }, 	// 116
        { "Unused",			                "MAP01"            }, 	// 117
        { "Unused",			                "MAP01"            }, 	// 118
        { "Unused",			                "MAP01"            }, 	// 119
        { "Unused",			                "MAP01"            }, 	// 120
        { "Unused",			                "MAP01"            }, 	// 121
        { "Unused",			                "MAP01"            }, 	// 122
        { "Unused",			                "MAP01"            }, 	// 123
        { "Unused",			                "MAP01"            }, 	// 124
        { "Unused",			                "MAP01"            }, 	// 125
        { "Unused",			                "MAP01"            }, 	// 126
        { "Unused",			                "MAP01"            }, 	// 127
        { "Unused",			                "MAP01"            }, 	// 128

        // Single Levels
		{ "The Spire",				        "TSP01"            }, 	// 129
		{ "The Eye",				        "EYE01"            }, 	// 130
        { "Sens",			                "HP103"            }, 	// 131
        { "A Clear Line Drawn",	            "AAA01"            }, 	// 132
        { "The Spire 2",			        "TSP201"           }, 	// 133
        { "Unused",			                "MAP01"            }, 	// 134
        { "Unused",			                "MAP01"            }, 	// 135
        { "Unused",			                "MAP01"            }, 	// 136
        { "Unused",			                "MAP01"            }, 	// 137
        { "Unused",			                "MAP01"            }, 	// 138
        { "Unused",			                "MAP01"            }, 	// 139
        { "Unused",			                "MAP01"            }, 	// 140
        { "Unused",			                "MAP01"            }, 	// 141
        { "Unused",			                "MAP01"            }, 	// 142
        { "Unused",			                "MAP01"            }, 	// 143
        { "Unused",			                "MAP01"            }, 	// 144
        { "Unused",			                "MAP01"            }, 	// 145
        { "Unused",			                "MAP01"            }, 	// 146
        { "Unused",			                "MAP01"            }, 	// 147
        { "Unused",			                "MAP01"            }, 	// 148
        { "Unused",			                "MAP01"            }, 	// 149
        { "Unused",			                "MAP01"            }, 	// 150
        { "Unused",			                "MAP01"            }, 	// 151
        { "Unused",			                "MAP01"            }, 	// 152
        { "Unused",			                "MAP01"            }, 	// 153
        { "Unused",			                "MAP01"            }, 	// 154
        { "Unused",			                "MAP01"            }, 	// 155
        { "Unused",			                "MAP01"            }, 	// 156
        { "Unused",			                "MAP01"            }, 	// 157
        { "Unused",			                "MAP01"            }, 	// 158
        { "Unused",			                "MAP01"            }, 	// 159
        { "Unused",			                "MAP01"            }, 	// 160
        { "Unused",			                "MAP01"            }, 	// 161
        { "Unused",			                "MAP01"            }, 	// 162
        { "Unused",			                "MAP01"            }, 	// 163
        { "Unused",			                "MAP01"            }, 	// 164
        { "Unused",			                "MAP01"            }, 	// 165
        { "Unused",			                "MAP01"            }, 	// 166
        { "Unused",			                "MAP01"            }, 	// 167
        { "Unused",			                "MAP01"            }, 	// 168
        { "Unused",			                "MAP01"            }, 	// 169
        { "Unused",			                "MAP01"            }, 	// 170
        { "Unused",			                "MAP01"            }, 	// 171
        { "Unused",			                "MAP01"            }, 	// 172
        { "Unused",			                "MAP01"            }, 	// 173
        { "Unused",			                "MAP01"            }, 	// 174
        { "Unused",			                "MAP01"            }, 	// 175
        { "Unused",			                "MAP01"            }, 	// 176
        { "Unused",			                "MAP01"            }, 	// 177
        { "Unused",			                "MAP01"            }, 	// 178
        { "Unused",			                "MAP01"            }, 	// 179
        { "Unused",			                "MAP01"            }, 	// 180
        { "Unused",			                "MAP01"            }, 	// 181
        { "Unused",			                "MAP01"            }, 	// 182
        { "Unused",			                "MAP01"            }, 	// 183
        { "Unused",			                "MAP01"            }, 	// 184
        { "Unused",			                "MAP01"            }, 	// 185
        { "Unused",			                "MAP01"            }, 	// 186
        { "Unused",			                "MAP01"            }, 	// 187
        { "Unused",			                "MAP01"            }, 	// 188
        { "Unused",			                "MAP01"            }, 	// 189
        { "Unused",			                "MAP01"            }, 	// 190
        { "Unused",			                "MAP01"            }, 	// 191
        { "Unused",			                "MAP01"            }, 	// 192
    };

	#if 1

		#include	"lex_util.c"
		#include	"lex_debug.c"
		#include	"lex_hub.c"
		#include	"lex_musicbox.c"
		#include	"lex_level.c"
		#include	"lex_voteskip.c"
		#include 	"lex_sharekey.c"

	#endif


}
























